src/main/java/edu/kis/powp/jobs2d/command/CompoundCommand.java

```diff
@@ -54,6 +54,35 @@ public List<DriverCommand> getCommands() {
         return commands;
     }
 
+    public boolean checkBoundaries()
```

Comment:
The visitor helps to avoid burdening classes with a new responsibilities (now the design is not SRP).